cirq-core/cirq/devices/grid_qubit.py

```diff
@@ -38,7 +38,36 @@ class _BaseGridQid(ops.Qid):
 
     def __hash__(self) -> int:
         if self._hash is None:
-            self._hash = hash((self._row, self._col, self._dimension))
+            # This approach seems to perform better than traditional "random" hash in `Set`
+            # operations for typical circuits, as it reduces bucket collisions. Caveat: it does not
```

Comment:
How did you evaluate this reduction in bucket collisions? Would be good to show this explicitly before we decide to abandon the standard tuple hash.

Reply:
Test code is up in the description. It's about 50% faster with this implementation.

One note is that it seems like it's only faster for copy-on-change ops like `s = s.union({q})`. It doesn't seem to have any effect when we operate on sets mutably like `s |= {q}`.